llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -837,16 +840,42 @@ void VPlanTransforms::optimizeForVFAndUF(VPlan &Plan, ElementCount BestVF,
       !SE.isKnownPredicate(CmpInst::ICMP_ULE, TripCount, C))
     return;
 
```

Comment:
Say something about what is about to happen now.

Reply:
Added, thanks